tests/test_other.py

```diff
@@ -1524,6 +1524,53 @@ def test(link_flags, lib_suffix):
     test(['-lfile'], '') # -l, auto detection from library path
     test([self.in_dir('libdir', 'libfile.so.3.1.4.1.5.9')], '.3.1.4.1.5.9') # handle libX.so.1.2.3 as well
 
+  def test_dynamic_link_module_heap_in_pthread(self):
+    # Test that a side module uses the same heap for global objects across all threads
+
+    # A side module with a global object with a constructor.
+    # The global object must have a constructor to make sure
+    # verify we construct it only once.
+    create_test_file('side.cpp', r'''
+      struct Data {
+          Data() : value(42) {}
+          int value;
+      } data;
+      int & get_value() {
+          return data.value;
+      }
+      ''')
+    self.run_process([
+      EMCC,
+      '-o', 'side.wasm',
+      'side.cpp',
+      '-pthread', '-Wno-experimental',
+      '-s', 'SIDE_MODULE=1'])
+
+    create_test_file('main.cpp', r'''
+      #include <stdio.h>
+      #include <thread>
+      int & get_value();
```

Comment:
How about calling this `get_address()` and returning a pointer .. it seems a little more descirptive.